app/config.py

```diff
@@ -0,0 +1,128 @@
+from functools import partial
+
+from envparse import Env, ConfigurationError
+
+from app.session import generate_new_key
+
+
+class Config(dict):
+
+    def from_object(self, obj):
+        for key in dir(obj):
+            if key.isupper():
+                config = getattr(obj, key)
+                if config is None:
+                    raise ConfigurationError(f'{key} not set')
+                self[key] = config
+
+    def __getattr__(self, name):
+        try:
+            return self[name]
+        except KeyError:
+            raise AttributeError('"{}" not found'.format(name))
+
+    def __setattr__(self, name, value):
+        self[name] = value
+
+
+class BaseConfig:
+    env = Env()
+    env = partial(env, default=None)
+
+    HOST = env("HOST")
+    PORT = env("PORT")
+    LOG_LEVEL = env("LOG_LEVEL")
+    STATIC_ROOT = "app/static"
+
+    ACCOUNT_SERVICE_URL = env("ACCOUNT_SERVICE_URL")
+    EQ_URL = env("EQ_URL")
+    JSON_SECRET_KEYS = env("JSON_SECRET_KEYS")
+
+    CASE_URL = env("CASE_URL")
+    CASE_AUTH = (env("CASE_USERNAME"), env("CASE_PASSWORD"))
+
+    COLLECTION_EXERCISE_URL = env("COLLECTION_EXERCISE_URL")
+    COLLECTION_EXERCISE_AUTH = (env("COLLECTION_EXERCISE_USERNAME"), env("COLLECTION_EXERCISE_PASSWORD"))
+
+    COLLECTION_INSTRUMENT_URL = env("COLLECTION_INSTRUMENT_URL")
+    COLLECTION_INSTRUMENT_AUTH = (env("COLLECTION_INSTRUMENT_USERNAME"), env("COLLECTION_INSTRUMENT_PASSWORD"))
+
+    IAC_URL = env("IAC_URL")
+    IAC_AUTH = (env("IAC_USERNAME"), env("IAC_PASSWORD"))
+
+    SAMPLE_URL = env("SAMPLE_URL")
+    SAMPLE_AUTH = (env("SAMPLE_USERNAME"), env("SAMPLE_PASSWORD"))
+
+    SURVEY_URL = env("SURVEY_URL")
+    SURVEY_AUTH = (env("SURVEY_USERNAME"), env("SURVEY_PASSWORD"))
+
+    SECRET_KEY = env("SECRET_KEY")
+
+
+class DevelopmentConfig:
+    env = Env()
+    HOST = env.str("HOST", default="0.0.0.0")
+    PORT = env.int("PORT", default="9092")
+    LOG_LEVEL = env("LOG_LEVEL", default="INFO")
+    STATIC_ROOT = "app/static"
+
+    ACCOUNT_SERVICE_URL = env.str("ACCOUNT_SERVICE_URL", default="http://0.0.0.0:9092")
```

Comment:
@jcox-dev I think this is supposed to be localhost? for all of them?

Reply:
Can be either I think 🤷‍♂️